app/models/concerns/rule_remediation.rb

```diff
@@ -11,10 +11,14 @@ def remediation_issue_id
   private
 
   def short_profile_ref_id
-    profile = profiles.canonical.first
-    short_ref_id = profile.ref_id.downcase.split(
+    profile_ref_ids = profiles.collect(&:ref_id).to_set
```

Comment:
Why not use a `[0]` at the end and assign only one ref_id as `profile_ref_id`?

If it's `nil` it can raise the error and it would avoid the `.first`.

Or am I missing something?

Reply:
`Set` isn't addressable. `.first` would work here, but the check (for non-uniqe refs) would not be possible.

Reply:
Why not use `.uniq.first` instead of transforming it to a set? collect should already return an array.

Reply:
I guess a remnant from Python idioms. The result is the same.